ambuda/models/auth.py

```diff
@@ -81,6 +81,40 @@ def __repr__(self):
         return f"<Role({self.id}, {self.name!r})>"
 
 
+class UserStatusLog(AmbudaUserMixin, Base):
+    """Tracks changes to user statuses."""
+
+    __tablename__ = "user_status_log"
+
+    #: Primary key.
+    id = pk()
+
+    #: The user whose status was changed.
+    user_id = Column(Integer, ForeignKey("users.id"), nullable=False)
+
+    #: Timestamp at which this status change occured.
+    timestamp = Column(DateTime, default=datetime.utcnow, nullable=False)
+
+    #: Describes the status change that occurred.
+    change_description = Column(String, nullable=False)
+
+    #: When should this status change expire/revert, defaults to never.
+    expiry = Column(DateTime, default=None, nullable=True)
```

Comment:
Since the semantics here are all in `change_description` which is unstructured text, there's nothing actionable we can do even if we know `expiry`. This seems like something we could fold into `change_description` itself